optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/util/ConcurrentCache.java

```diff
@@ -0,0 +1,31 @@
+package org.optaplanner.core.impl.domain.solution.util;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Thread-safe basic cache implementation.
+ *
+ * @param <K> the type of keys for the cache
+ * @param <V> the type of values stored in the cache
+ * */
+public final class ConcurrentCache<K, V> extends ConcurrentHashMap<K, V> {
```

Comment:
Renaming discussion:
- ConcurrentCache sounds like a standard class from the JDK. Can we make it less general?
- We use it to store calculations, not retrieve information for a remote service/database.
- We use this Cache as a lazy initialization only: once a value is calculated, it is never recalculated.
- After discussion with Mario, he rightfully suggested this is a Memoization, see https://en.wikipedia.org/wiki/Memoization

As most people don't know what a Memoization is, shall we call it MemoizationCache? :) I'd leave out the name Concurrent, as that seems implicit.

Reply:
Sounds good